src/protocols/secure_channel/MessageCounterManager.h

```diff
@@ -122,30 +133,29 @@ class MessageCounterSyncMgr : public Messaging::ExchangeDelegate
      */
     struct ReceiveTableEntry
     {
-        System::PacketBufferHandle msgBuf; /**< A handle to the PacketBuffer object holding
-                                                the message data. This is non-null if and only
-                                                if this entry is in use. */
+        ReceiveTableEntry() : peerNodeId(kUndefinedNodeId) {}
+
+        NodeId peerNodeId;                  /**< The peerNodeId of the message. kUndefinedNodeId if is not in use. */
+        PacketHeader packetHeader;          /**< The packet header for the message. */
```

Comment:
Same questions here: how big is the receive table before this change?  How big after?  There were earlier review comments asking for this to not be regressed by continuing to store the packet header in the buffer that seem to have been ignored.

Also, nodeid is not a good way to identify things here, because it's not going to be well-defined in multi-fabric.  It's also rather large.   If we absolutely need to store something here that we can extract the node id from, what is the smallest thing we can store?

Reply:
I have add an issue to address this #6340, it will be fixed afterward.

Reply:
I guess I can live with that, though it's pretty weird that we already have a working system for not having this problem and this PR is _removing_ that only to add it back later....